### PR TITLE
Fix/ghg top 10 emitters

### PR DESCRIPTION
**Pull request by Bluesmile82**
WIP: Legend style fix needed from a CW components PR: https://github.com/ClimateWatch-Vizzuality/climate-watch-components/pull/80

This PR adds the top 10 emitters to provinces instead of All selected, and also the target emissions

Note: Only Indonesia province has data. Also the data has to be updated as there are some problems with the units and metrics. You can see that the total value is not right

Extra: Refactored the selectors in different files. Let me know what you think about it. It could be some standard for the charts. The historical-emissions-filter-selectors could be also divided but I thought it was too much. Also we could extract some helpers and selectors to reuse them in other charts.

image

image

### CODE REVIEW

**Comment by Bluesmile82** [comment]
Finally ready! Sorry for the big PR

**Comment by simaob** [comment]
lgtm =) Let's get it reviewed by the front enders and merged if all good on Monday! =)

**Comment by aabdaab** [comment]
I think there's one small bug: on initial render instead of having 10 top emitters chosen and having their names in the legend we have a `Top 10 emitters` label in the legend and they're not picked in the dropdown, see below:
top10emitters

File: app/javascript/app/providers/ghg-target-emissions-provider/ghg-target-emissions-provider-actions.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+);
+export const fetchGHGTargetEmissionsFail = createAction(
+  'fetchGHGTargetEmissionsFail'
+);
+
+export const fetchGHGTargetEmissions = createThunkAction(
+  'fetchGHGTargetEmissions',
+  params => (dispatch, state) => {
+    const { GHGTargetEmissions } = state();
+    if (!GHGTargetEmissions.loading) {
```
Comment (aabdaab) [inline app/javascript/app/providers/ghg-target-emissions-provider/ghg-target-emissions-provider-actions.js]: I think it's also worth checking if the data is already fetched so prevent re-fetching

**Comment by aabdaab** [comment]
we could actually do the same with `actions`; to have a dedicated folder for commonly used actions like `updateFilters`

**Comment by Bluesmile82** [comment]
To have a selectors folder will make sense. Although this GHG chart can have many specifics. Sorry if the differences are not clear at plain sight. So I split these selectors into different folders because we usually end up with a huge file with all the functionality and selectors for the chart and is quite difficult to debug and change. These are split regarding their use. 
The historical-emissions-selectors is the final file that groups all selectors and works as an entry file for the rest of the files,

The historical-emissions-get-selectors contains mostly functions and maybe some selectors that get information from the state to use them in other selectors.

The filter selectors are specific to the dropdowns in the chart. They are subdivided into options, defaults, and selected values. There are some helper selectors like getFieldOptions, and functions like findOption that help to reduce the lines of code.

The data selectors are made for the chart data and configuration. Most of the logic of the chart occurs here.

Finally, the fetch selectors are made for the parameters that are passed to the provider. 

Let me know if you have some suggestions or better names

**Comment by weberjavi** [comment]
Thanks for explaining it!!
It really makes sense ✊

File: app/javascript/app/providers/ghg-target-emissions-provider/ghg-target-emissions-provider-actions.js (hunk 1 above)
Comment (Bluesmile82) [reply app/javascript/app/providers/ghg-target-emissions-provider/ghg-target-emissions-provider-actions.js]: I'm checking in the action if the params are the same. The params change the data we store in this case so we can't prevent the fetching

**Comment by simaob** [comment]
If https://github.com/ClimateWatch-Vizzuality/indonesia-platform/pull/29 is based on this PR shall we close this one?

**Pull request closed without merging**